client/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Axios from 'axios';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hostel: [],
+    };
+  }
+
+  componentDidMount() {
+    this.getHostels();
+  }
+
+  getHostels() {
+    Axios.get('api/house/', {
+      params: {
+        id: 50,
+      },
+    })
+      .then((res) => {
+        console.log(res, 'res in getHostels')
+        this.setState({ hostel: res.body });
+        this.componentDidMount();
+      })
+      .catch((err) => err);
```

Comment:
A few things of note. You shouldn't directly call lifecycle methods in React. You can modify what they do when something happens (such as getting hostels when the component mounts). But if you feel the need to call it, you should probably modify the way your functions are working. Second, if we receive an error, what are we doing? Should we notify the user that something went wrong?

Reply:
>You shouldn't directly call lifecycle methods in React. You can modify what they do when something happens (such as getting hostels when the component mounts). But if you feel the need to call it, you should probably modify the way your functions are working. 

I don't remember exactly what I was attempting to do and/or test with that at the time, but it was essentially a quick "fix" to test something. I has since been fixed and lifecycle methods are not being called directly any more.

> Second, if we receive an error, what are we doing? Should we notify the user that something went wrong?

I know that for certain errors, it makes sense to notify the user which is why we see 403 and 404 error pages on websites, but for projects like this where it doesn't make sense to build out an error page, is it a better idea to fail silently or to use something like `alert('blah, blah, blah... there was an error...')`?